### DISCUSSION TITLE
Installing Freetype

### DISCUSSION THREAD

**Discussion by rsztype** [fonttools/fontbakery#3978] [Q&A] [answered]
Hello i'm having this issues.
I'm running python on Monterey iMac 2015.

`
gs-iMac:~ giuseppesalerno$ python -m pip install --upgrade fontbakery'[freetype]'
Collecting fontbakery[freetype]
  Could not fetch URL https://pypi.python.org/simple/fontbakery/: There was a problem confirming the ssl certificate: [SSL: TLSV1_ALERT_PROTOCOL_VERSION] tlsv1 alert protocol version (_ssl.c:661) - skipping
  Could not find a version that satisfies the requirement fontbakery[freetype] (from versions: )
No matching distribution found for fontbakery[freetype]`

**Comment by miguelsousa** [comment]
Does it work if you remove the single quotes?

    python -m pip install --upgrade fontbakery[freetype]

**Comment by rsztype** [comment]
unfortunately not

`gs-iMac:~ giuseppesalerno$ python -m pip install --upgrade fontbakery[freetype]
Collecting fontbakery[freetype]
  Could not fetch URL https://pypi.python.org/simple/fontbakery/: There was a problem confirming the ssl certificate: [SSL: TLSV1_ALERT_PROTOCOL_VERSION] tlsv1 alert protocol version (_ssl.c:661) - skipping
  Could not find a version that satisfies the requirement fontbakery[freetype] (from versions: )
No matching distribution found for fontbakery[freetype]
`

**Comment by miguelsousa** [comment]
Do you mind trying the command without the `[freetype]` part? That will tell us if there a problem with how Font Bakery extras are requested, or if it's something else.

**Comment by rsztype** [comment]
Sure, i did.
Same error.

**Comment by miguelsousa** [comment]
Thanks. That suggests that there's something wrong with your system that may not be specific to fontbakery.
Try something as basic as the command below, which updates pip itself. You can use any other package available at pypi.org if you prefer. 

    python -m pip install --update pip

**Comment by rsztype** [comment]
it's strange coz i'm using Font Bakery for checking and works fine.

**Comment by rsztype** [comment]
`gs-iMac:~ giuseppesalerno$ python3 --version
Python 3.11.0
gs-iMac:~ giuseppesalerno$ 


`

**Comment by miguelsousa** [comment]
> it's strange coz i'm using Font Bakery for checking and works fine.

What works fine? I'm confused now

**Comment by rsztype** [comment]
no unfortunately

`gs-iMac:~ giuseppesalerno$ python -m pip install --update pip

Usage:   
  /Library/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python -m pip install [options] <requirement specifier> [package-index-options] ...
  /Library/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python -m pip install [options] -r <requirements file> [package-index-options] ...
  /Library/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python -m pip install [options] [-e] <vcs project url> ...
  /Library/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python -m pip install [options] [-e] <local project path> ...
  /Library/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python -m pip install [options] <archive url/path> ...

no such option: --update
`

**Comment by miguelsousa** [comment, marked as answer]
OK, I see what's going on. You have both Python 2.7 and 3.11 installed. In your system the `python` command will invoke 2.7, which Font Bakery no longer supports. So in your case the installation command should be,

    python3 -m pip install --upgrade fontbakery'[freetype]'

BUT because you have 3.11 installed the command will likely fail because Font Bakery depends on the `lxml` package which at the moment hasn't published a wheel for 3.11.

**Comment by miguelsousa** [comment]
> no such option: --update

My bad, the correct option is `--upgrade`

**Comment by rsztype** [comment]
Works!

Thank you so much

**Comment by kalapi** [comment]
I'm getting the following Freetype error when running `FontBakery 0.9.2` on `Python 3.8`

```
ERROR Failed with RuntimeError: Freetype library not found                  
          ↳ File                                                                
          "/Users/UT-development/Documents/_Programming/testing_environments/FMa
          ke37/lib/python3.8/site-packages/fontbakery/checkrunner.py", line 183,
          in _exec_check                                                        
          for sub_result in result:  # Might raise.                             
          ↳ File                                                                
          "/Users/UT-development/Documents/_Programming/testing_environments/FMa
          ke37/lib/python3.8/site-packages/fontbakery/profiles/universal.py",   
          line 1726, in com_adobe_fonts_check_freetype_rasterizer               
          import freetype                                                       
          ↳ File                                                                
          "/Users/UT-development/Documents/_Programming/testing_environments/FMa
          ke37/lib/python3.8/site-packages/freetype/init.py", line 24, in       
          from freetype.raw import *                                            
          ↳ File                                                                
          "/Users/UT-development/Documents/_Programming/testing_environments/FMa
          ke37/lib/python3.8/site-packages/freetype/raw.py", line 52, in        
          raise RuntimeError('Freetype library not found')                      


    Result: ERROR
```

I tried installing/upgrading FontBakery using the command in @miguelsousa's comment and I get this line during install:

```
WARNING: fontbakery 0.9.2 does not provide the extra 'freetype'
```

Also getting this weird issue with trying to re-install `freetype-py`, maybe this is the issue?

```
fontbakery 0.9.2 requires freetype-py!=2.4.0, but you have freetype-py 2.4.0 which is incompatible.
```
